### PR TITLE
useNotifyOnNewEpoch, some small QoL additions to useService (isRunning, .etc)

### PR DESCRIPTION
**Pull request by truemiller**
## Proposed changes

Fix lint errors on useNotifyOnNewEpoch
Added some deployment status related booleans for ease of use

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/client/enums.ts [hunk 1]
```diff
@@ -30,6 +30,24 @@ export enum MiddlewareDeploymentStatus {
...
+  MiddlewareDeploymentStatus.STOPPING,
+];
+
+/** @node statuses where middleware deployment is running */
+export const MiddlewareRunningStatuses = [
+  MiddlewareDeploymentStatus.DEPLOYED,
+  ...MiddlewareTransitioningStatuses,
+];
+
+/** @note statuses where middleware is in the process of building/creating a new deployment */
```
Comment (Tanya-atatakai) [inline frontend/client/enums.ts:45]: can ignore, but the comment is a bit confusing. E.g. `MiddlewareDeploymentStatus.DEPLOYING` sounds more like a status representing `in the process of building/creating`

File: frontend/hooks/useService.ts [hunk 2]
```diff
@@ -139,6 +143,21 @@ export const useService = ({
...
+    : false;
+
+  /** @note deployment is running, or transitioning, both assume the deployment is active */
+  const isServiceRunning = deploymentStatus
+    ? MiddlewareRunningStatuses.includes(deploymentStatus)
+    : false;
+
+  /** @note new deployment being created/built */
+  const isServiceBuilding = deploymentStatus
+    ? MiddlewareTransitioningStatuses.includes(deploymentStatus)
```
Comment (Tanya-atatakai) [inline frontend/hooks/useService.ts:158]:
```suggestion
    ? MiddlewareBuildingStatuses.includes(deploymentStatus)
```

File: frontend/client/enums.ts (hunk 1 above)
Comment (truemiller) [reply frontend/client/enums.ts:45]: agreed, however it represents "starting" from both built and stopped

File: frontend/hooks/useService.ts (hunk 2 above)
Comment (truemiller) [reply frontend/hooks/useService.ts]: thanks!

**Pull request merged**